### ISSUE TITLE
unsymmetrical results

### ISSUE THREAD

**Issue by yoavmil** [geometryprocessing/voroffset#1]
TL;DR
for an symmetrical mesh, the results isn't symmetric at all axes.

long version
I came across your library, to find a solution for filtering small features out of a mesh, for 3D printing.
To test you library, I made a cube with a sphere cavity in it:
![image](https://user-images.githubusercontent.com/5158716/82306702-0f742400-99c8-11ea-9570-6eeab40344f1.png)
I ran erosion on it, to make the walls thinner, which worked correctly. See pictures:
![image](https://user-images.githubusercontent.com/5158716/82307404-fc158880-99c8-11ea-9329-eddd2be73274.png)
![image](https://user-images.githubusercontent.com/5158716/82307657-51ea3080-99c9-11ea-9692-5a4558a98111.png)

Then I ran dilation on it, to bring it back to the original size, and the resulting mesh wasn't as expected. The overall bounding box wasn't as the original bounding box, and moreover, the Z axis grew back to originals size, but the XY sizes remaind as after the erosion. Also the holes aren't the same at all directions, see picture:

![image](https://user-images.githubusercontent.com/5158716/82308401-67138f00-99ca-11ea-91a8-8516319e095a.png)
VS
![image](https://user-images.githubusercontent.com/5158716/82308420-6da20680-99ca-11ea-9850-f6963ec22665.png)

Please advise!

**Comment by jdumas** [comment]
Hi,

Thanks for trying out this library!

The problem you're describing with the dilation seems to be because you didn't add enough padding to the XY grid, so the cube gets truncated to the original XY bbox. Try adding `--padding R` when running the script.

That being said, It is expected that the dilation/erosion results are not exactly the same along XY than they are along the Z axis. If you read [our publication](https://www.jdumas.org/publication/2019/voroffset/) you should understand why that is. Our method works best when targeting voxel data/3D printing/or when you know the XY resolution is limited.

**Comment by yoavmil** [comment]
The padding worked. Result is symmetric.
I have another question, but I'll ask in another issue

**Comment by yoavmil** [comment]
how ever, to make this tool resilient, it would be good if the padding was done automatically.

**Comment by jdumas** [comment]
That's a good idea. Not sure when I'll have time to do this though.
